google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -83,6 +85,34 @@
             + "Please explicitly set your Project ID in FirestoreOptions.");
     this.databasePath =
         ResourcePath.create(DatabaseRootName.of(options.getProjectId(), options.getDatabaseId()));
+
+    if (!options.areTimestampsInSnapshotsEnabled()) {
+      LOGGER.warning(
+          "The behavior for java.util.Date objects stored in Firestore is going to change "
+              + "AND YOUR APP MAY BREAK.\n"
+              + "To hide this warning and ensure your app does not break, you need to add "
+              + "the following code to your app before calling any other Cloud Firestore "
+              + "methods:\n"
+              + "\n"
+              + "FirestoreOptions options = \n"
+              + "  FirestoreOptions.newBuilder().setTimestampsInSnapshotsEnabled(true).build();\n"
+              + "Firestore firestore = options.getService();\n"
+              + "\n"
+              + "With this change, timestamps stored in Cloud Firestore will be read back as "
+              + "com.google.cloud.Timestamp objects instead of as system java.util.Date "
+              + "objects. So you will also need to update code expecting a java.util.Date to "
+              + "instead expect a Timestamp. For example:\n"
+              + "\n"
+              + "// Old:\n"
+              + "java.util.Date date = snapshot.getDate(\"created_at\");\n"
+              + "// New:\n"
+              + "Timestamp timestamp = snapshot.getTimestamp(\"created_at\");\n"
+              + "java.util.Date date = timestamp.toDate();\n"
```

Comment:
IIUC the reason we nee this setting is that there are methods that "formally" return `Object` and this will change whether the object is actually a `Date` or `Timestamp` right? I think it's worth giving an example of such a method.

Reply:
Yeah, that's a good point. I updated it to use `DocumentSnapshot.get()`.